hudi-spark-datasource/hudi-spark3.3.x/src/main/scala/org/apache/spark/sql/hudi/catalog/HoodieCatalog.scala

```diff
@@ -355,7 +355,7 @@ object HoodieCatalog {
         identityCols += col
 
 
-      case BucketTransform(numBuckets, FieldReference(Seq(col))) =>
+      case BucketTransform(numBuckets, Seq(FieldReference(Seq(col))), _) =>
         bucketSpec = Some(BucketSpec(numBuckets, col :: Nil, Nil))
```

Comment:
[SPARK-37627](https://github.com/apache/spark/pull/34914) API change. 
This API hasn't changed a lot but this code change is the only way I found to make compiler happy. Please review this